packages/core-data/src/hooks/use-resource-permissions.ts

```diff
@@ -152,7 +172,7 @@ export default function useResourcePermissions< IdType = void >(
 
 export function __experimentalUseResourcePermissions(
 	resource: string,
-	id?: IdType
+	id?: unknown
```

Comment:
Is that change necessary? The type of the stable method didn't change.

Reply:
The stable method has a generic `IdType` parameter while this one doesn't.

Reply:
I really need to learn more about TypeScript 🙈 

Feel free to land this PR 😄